homeassistant/components/medcom_ble/config_flow.py

```diff
@@ -0,0 +1,164 @@
+"""Config flow for Medcom BlE integration."""
+
+from __future__ import annotations
+
+import dataclasses
+import logging
+from typing import Any
+
+from bleak import BleakError
+from bluetooth_data_tools import human_readable_name
+from medcom_ble import MedcomBleDevice, MedcomBleDeviceData
+import voluptuous as vol
+
+from homeassistant.components import bluetooth
+from homeassistant.components.bluetooth import (
+    BluetoothServiceInfo,
+    async_discovered_service_info,
+)
+from homeassistant.config_entries import ConfigFlow
+from homeassistant.const import CONF_ADDRESS
+from homeassistant.data_entry_flow import FlowResult
+
+from .const import DOMAIN
+
+_LOGGER = logging.getLogger(__name__)
+
+
+@dataclasses.dataclass
+class Discovery:
+    """A discovered bluetooth device."""
+
+    name: str
+    discovery_info: BluetoothServiceInfo
+    device: MedcomBleDevice
+
+
+def get_name(device: MedcomBleDevice) -> str:
+    """Generate name with model and identifier for device."""
+
+    name = device.friendly_name()
+    if identifier := device.identifier:
+        name += f" ({identifier})"
+    return name
+
+
+class InspectorBLEConfigFlow(ConfigFlow, domain=DOMAIN):
+    """Handle a config flow for Medcom BLE radiation monitors."""
+
+    VERSION = 1
+
+    def __init__(self) -> None:
+        """Initialize the config flow."""
+        self._discovery_info: BluetoothServiceInfo | None = None
+        self._discovered_device: Discovery | None = None
+        self._discovered_devices: dict[str, Discovery] = {}
+
+    async def _get_device_data(
+        self, discovery_info: BluetoothServiceInfo
+    ) -> MedcomBleDevice:
+        ble_device = bluetooth.async_ble_device_from_address(
+            self.hass, discovery_info.address
+        )
+        if ble_device is None:
+            _LOGGER.debug("no ble_device in _get_device_data")
+            raise BleakError("Cannot find BLE device")
```

Comment:
This doesn't look like a bleakerror.

Reply:
Updating to raise an AbortFlow error, since this is essentially the outcome of not finding the BLE device...